pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/plan/server/ServerPlanRequestUtils.java

```diff
@@ -255,7 +255,8 @@ private static List<Expression> computeInOperands(List<Object[]> dataContainer,
         }
         Arrays.sort(arrFloat);
         for (int rowIdx = 0; rowIdx < numRows; rowIdx++) {
-          expressions.add(RequestUtils.getLiteralExpression(arrFloat[rowIdx]));
+          // TODO: Create float literal when it is supported
+          expressions.add(RequestUtils.getLiteralExpression(Double.parseDouble(Float.toString(arrFloat[rowIdx]))));
```

Comment:
is this even needed? we should handle it in RequestUtils 
- AFAIK the arrFloat[rowIdx] will point to the double signature, 
- once uncomment the TODO it will use the float signature?

Reply:
This is needed. We cannot directly cast `float` to `double` because that will create wrong value (e.g `0.05f` -> `0.05000000074505806`), then cause wrong result. Added more notes